### PR TITLE
[WIP] Update NearBridge to the latest LightClient spec

### PR DESCRIPTION
**Pull request by k06a**

### CODE REVIEW

File: libs-sol/nearbridge/contracts/NearBridge.sol [hunk 1]
```diff
@@ -123,32 +125,23 @@ contract NearBridge is Ownable {
...
-        //         nearBlock.next_bps.validatorStakes
-        //     ),
-        //     "NearBridge: Less than 2/3 voted by the block after next"
-        // );
+        // 4. approvals_after_next contain signatures that check out against the block producers for the epoch of the block
+        // 5. The signatures present in approvals_after_next correspond to more than 2/3 of the total stake
+        uint256 totalStake = 0;
+        uint256 votedFor = 0;
+        if (prev.bps_count > 0) {
+            require(nearBlock.next_bps.validatorStakes.length == prev.bps_count, "NearBridge: number of BPs should match number of approvals");
```
Comment (SkidanovAlex) [inline libs-sol/nearbridge/contracts/NearBridge.sol:133]: As per spec, number of approvals can be smaller than the number of BPs

File: libs-sol/nearbridge/contracts/NearBridge.sol [hunk 2]
```diff
@@ -123,32 +125,23 @@ contract NearBridge is Ownable {
...
+        }
+        for (uint i = 0; i < nearBlock.next_bps.validatorStakes.length; i++) {
+            totalStake = totalStake.add(
+                nearBlock.next_bps.validatorStakes[i].stake
+            );
+            if (!nearBlock.approvals_after_next[i].none) {
+                // Assume presented signatures are valid, but this could be challenged
+                votedFor = votedFor.add(nearBlock.next_bps.validatorStakes[i].stake);
```
Comment (SkidanovAlex) [inline libs-sol/nearbridge/contracts/NearBridge.sol:141]: Seems like we are using block producers for the **next** epoch?

**Comment by k06a** [comment]
Fixed #47

File: libs-sol/nearbridge/contracts/NearBridge.sol (hunk 2 above)
Comment (k06a) [reply libs-sol/nearbridge/contracts/NearBridge.sol:140]: Will fix that

File: libs-sol/nearbridge/contracts/NearBridge.sol (hunk 1 above)
Comment (k06a) [reply libs-sol/nearbridge/contracts/NearBridge.sol:132]: Some approvals are empty but lengths should match, isn't it?

Comment (SkidanovAlex) [reply libs-sol/nearbridge/contracts/NearBridge.sol:132]: No, the length can be shorter, in which case it is assumed that the rest is `None`s.

It is not very noticeable in the spec, but it's there:

> approvals_after_next can contain fewer elements than next_bps in the previous epoch.

(https://github.com/nearprotocol/NEPs/blob/master/specs/ChainSpec/LightClient.md)

**Pull request merged**